### DISCUSSION TITLE
Which style of `hunk-header=syntax` with `color-only` is preferable?

### DISCUSSION THREAD

**Discussion by ghost** [dandavison/delta#442] [Ideas]
https://github.com/dandavison/delta/pull/437#issuecomment-740691252

@dandavison asked me to achieve `hunk-header=syntax` with `color-only`, but there are two options to achieve.

## 1: without @@
It's easy but needs to change a test into complicated way. Also, I think it's less viewable.
<img width="796" alt="ss 3" src="https://user-images.githubusercontent.com/41639488/101766780-9fc1b400-3b26-11eb-9e15-db5acb1b323a.png">

## 2: with @@
I think this is preferred, easier to see.

<img width="766" alt="ss 2" src="https://user-images.githubusercontent.com/41639488/101766815-abad7600-3b26-11eb-9427-279f57b15d41.png">

But code will be messy in paint.rs. To implement this, need to avoid here.
https://github.com/dandavison/delta/blob/c71e3f4e90b21bb48f5b67c5c62da165573ceec9/src/paint.rs#L422-L424

## What I want to ask
In either way, I need to change code a bit complicated.

So, I'd like to decide which one to progress.

**Comment by ghost** [comment]
resolved here
https://github.com/dandavison/delta/pull/443

**Comment by dandavison** [comment]
Yes, I agree, I think it has to be (2). The reason I say "has to be" is that I think we want `color-only` to mean: if you remove the colors, then the remaining text is _identical_ to git's output throughout the entire patch: including the diff itself, and the metadata lines. So all the weird stuff like `---` and `@@` has to remain unchanged in delta's output.

Do you agree that that is what `color-only` should mean, and do you agree that we are currently obeying that rule?

**Comment by ghost** [comment]
> Do you agree that that is what color-only should mean, and do you agree that we are currently obeying that rule?

Yes, exactly! As you said, I'm obeying that rule. 
I'm relieved to hear that, thank you.
